scripts/code_style.py

```diff
@@ -163,14 +163,26 @@ def main() -> int:
                 + uncrustify_version + "' (Note: The only supported version" \
                 "is " + UNCRUSTIFY_SUPPORTED_VERSION + ")", file=STDOUT_UTF8)
 
-    src_files = get_src_files()
-
     parser = argparse.ArgumentParser()
-    parser.add_argument('-f', '--fix', action='store_true', \
-            help='modify source files to fix the code style')
+    parser.add_argument('-f', '--fix', action='store_true',
+                        help='modify source files to fix the code style')
+    # --files is almost useless: it only matters if there are no files
+    # ('code_style.py' without arguments checks all files known to Git,
+    # 'code_style.py --files' does nothing). 'code_style.py --files ...' is
+    # intended as a stable ("porcelain") way to restyle a possibly empty
+    # set of files.
+    parser.add_argument('--files', action='store_true',
```

Comment:
What's the use of this option if we can drop it and achieve the same results?

Reply:
Per the comment above, it handles the corner case of not passing any non-option arguments. This is useful for automated scripts that restyle a possibly-empty set of files.

Reply:
Ah, right. This use case wasn't obvious to me after reading the comment.